sql/auth/sql_user_table.cc

```diff
@@ -2060,14 +2060,17 @@ int open_grant_tables(THD *thd, TABLE_LIST *tables,
 #ifdef WITH_WSREP
 bool wsrep_check_system_user_privilege(THD *thd,
                                        const List<LEX_USER> &user_list) {
-  Acl_cache_lock_guard acl_cache_lock(thd, Acl_cache_lock_mode::READ_MODE);
 
-  if (!acl_cache_lock.lock()) {
-    return true;
-  }
+  if (WSREP(thd) && !thd->wsrep_applier) {
+    Acl_cache_lock_guard acl_cache_lock(thd, Acl_cache_lock_mode::WRITE_MODE);
```

Comment:
Taking the lock in write mode may not be necessary here as read mode will just do the job.